third_party/kcfinder/core/bootstrap.php

```diff
@@ -178,4 +178,196 @@ public function gc($maxlifetime) {
+    }
+ 
+    /**
+     * read() reads the session data from the Limesurvey DB session
+     * storage and returns the results
+     *
+     * @param string $sSessionId - the session identifier
+     * @return string - session data
+     */ 
+    public function read($sSessionId)
+    {
+        try 
+        { 
+            $oDbh = $this->getDbh(); 
+	    $sTblName = $this->sSessionTableName;
+	    
+            $sSql = "SELECT data "
+                    ."FROM $sTblName " 
+                    ."WHERE expire > :expire "
+	            ."AND id = :id";
+
+	    $oStmt = $oDbh->prepare($sSql);
+	    $iNow = time();
+            $oStmt->bindParam(':expire', $iNow, PDO::PARAM_INT);
+            $oStmt->bindParam(':id', $sSessionId, PDO::PARAM_STR);
+     
+            if ( $oStmt->execute() )
+	    {
+                $aRow = $oStmt->fetch(PDO::FETCH_ASSOC); 
+      	        return $aRow['data'];
+            }
+        }
+        catch (PDOException $e)
+        {
+            // nothing to do here
+        }
+ 
+        return '';
+    }
+
+    /**
+     * write() function is executed when the session is being updated.
+     *
+     * @param string $sSessionId - the session identifier
+     * @param string $sData - the session identifier
+     *
+     * @return bool - true on success
+     */ 
+    public function write($sSessionId, $sData) 
+    {
+        try 
+	{ 
+	    $oDbh = $this->getDbh(); 
+	    $sTblName = $this->sSessionTableName;
+
+	    $sSql = "UPDATE $sTblName "
+	           ."SET data=:data "
+                   ."WHERE id=:id";
+            $oStmt = $oDbh->prepare($sSql);
+            $oStmt->bindParam(':id', $sSessionId, PDO::PARAM_STR);
+            $oStmt->bindParam(':data', $sData, PDO::PARAM_STR);
+     
+            $oStmt->execute();
+            return true;
+        } 
+        catch (PDOException $e) 
+        {
+            //nothing to do here
+        }
+ 
+        return false;
+    }
+ 
+    /** destroy() called when the session should be destroyed. 
+     * Does nothing as sessions are managed by Limesurvey
+     *
+     * @param string $sSessionId - the session identifier
+     * @return bool - true on success
+     */
+    public function destroy($sessionId) 
+    {
+        return true;
+    }
+ 
+    /** gc() Should cleanup old sessions, not implemented
+     *  as sessions are managed by Limesurvey
+     *
+     * @param string $maxlifetime - session older than maxlifetime should be destroyed
+     * @return bool - true on success
+     */
+    public function gc($maxlifetime) 
+    {
+        return true;
+    }
+}
+
+ /** 
+  * checkLSSession() read Limesurvey configuration and if sessions are
+  * managed by CDbHttpSession create a new instance of
+  * LSSessionSaveHandler
+  *
+  * @author Gabriele Mambrini
+  */
+function checkLSSession() 
+{
+    //relative path calculated from the path where kcfinder is running
+    $sLimesurveyFolder = realpath( dirname(__FILE__) . "/../../../application");
```

Comment:
PS : i made the same error : asking why we don't use Yii model request .... but PDO